Source/sdc/sdc_newton_solve.H

```diff
@@ -322,35 +248,35 @@ sdc_newton_solve(const Real dt_m,
         dgesl<NumSpec+2>(Jac, ipvt, f_rhs);
 #endif
 
-        for (int n = 0; n < NumSpec+2; ++n) {
-            dU_react[n] = f_rhs(n+1);
+        for (int n = 1; n < NumSpec+1; ++n) {
+            dU_react(n) = f_rhs(n);
         }
 
         // how much of dU_react should we apply?
         Real eta = 1.0_rt;
-        for (int n = 0; n < NumSpec+2; ++n) {
-            dU_react[n] *= eta;
-            U_react[n] += dU_react[n];
+        for (int n = 1; n < NumSpec+1; ++n) {
+            U_react(n) += eta * dU_react(n);
         }
 
-        GpuArray<Real, NumSpec+2> eps_tot;
-
-        eps_tot[0] = tol_dens * std::abs(U_react[0]) + sdc_solver_atol;
+        Array1D<Real, 1, NumSpec+1> eps_tot;
 
         // for species, atol is the mass fraction limit, so we
         // multiply by density to get a partial density limit
+
         for (int n = 0; n < NumSpec; ++n) {
-            eps_tot[1 + n] = tol_spec * std::abs(U_react[1 + n]) + sdc_solver_atol * std::abs(U_react[0]);
+            eps_tot(1 + n) = tol_spec * std::abs(U_react(1 + n)) +
+                sdc_solver_atol * std::abs(U_new[URHO]);
         }
-        eps_tot[NumSpec+1] = tol_ener * std::abs(U_react[NumSpec+1]) + sdc_solver_atol;
+        eps_tot(NumSpec+1) = tol_ener * std::abs(U_react(NumSpec+1)) + sdc_solver_atol;
 
         // compute the norm of the weighted error, where the
         // weights are 1/eps_tot
+
         auto err_sum = 0.0_rt;
-        for (int n = 0; n < NumSpec+2; ++n) {
-            err_sum += dU_react[n] * dU_react[n] / (eps_tot[n]* eps_tot[n]);
+        for (int n = 1; n < NumSpec+1; ++n) {
```

Comment:
Is this correct or should it be n <= NumSpec+1?

Reply:
your correct.  It should be <=

Reply:
updated